examples/CIB/ex1/main.cpp

```diff
@@ -399,6 +495,83 @@ main(int argc, char* argv[])
             pout << "+++++++++++++++++++++++++++++++++++++++++++++++++++\n";
             pout << "\n";
 
+            // Fill ghost cells of pressure and velocity to compute hydrodynamic forces.
```

Comment:
Regarding cloned variables: the only reason to clone is to ensure that the data layout is the same. In this case, you already know the data layout (e.g. SideVariable with depth=1 for velocity, CellVariable with depth=1 for pressure), so you can just initialize appropriate velocity and pressure Variables internally in the class constructor.